### PR TITLE
feat: implements new dynamic configuration question logic

### PR DESCRIPTION
**Pull request by baranbartu**

### CODE REVIEW

File: cli/cli/dynamic_configuration/core.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
+import importlib
+
+import click
+
+
+def find_question_by_id(questions, question_id):
```
Comment (Younday) [inline cli/cli/dynamic_configuration/core.py:6]: Can possibly add some docstring to these  functions. However, the function names are quite clear on themselves as well. Just noticed that in the `predefined_funcs.py` file there are indeed some docstrings.

**Review by Younday — APPROVED** [review]
In general it looks good to me, just one comment on checking any other places to add docstring to. Other than that, I think this can be merged in (tests are there, and they even pass)

File: cli/cli/dynamic_configuration/core.py (hunk 1 above)
Comment (baranbartu) [reply cli/cli/dynamic_configuration/core.py:6]: ohh yes, sure. Will do it.

Comment (baranbartu) [reply cli/cli/dynamic_configuration/core.py:6]: done. @Younday

**Pull request merged**